server/src/extent_tree.c

```diff
@@ -344,7 +347,7 @@ int extent_tree_truncate(
         } else {
             /* the range of this node overlaps with the truncated size
              * so just update its end to be the new size */
-            node->end = size;
+            node->end = target_offset;
```

Comment:
We could then change this to be ```node->end = size -1;```

Reply:
Yes, you are right. Do we need to deallocate the extent tree when the size is zero?

Reply:
No, we can allow new extents to come in.  I think we only need to deallocate when destroying the inode.